tools/src/wyvern/target/corewyvernIL/support/GenContext.java

```diff
@@ -22,4 +63,21 @@ public static String generateName() {
 		return "var_"+(count++);
 	} 
 	private static int count = 0;
+
+	public GenContext rec(String newName, TypedAST ast) {
+		if(ast instanceof TypeVarDecl) {
+			String typeName = ((TypeVarDecl) ast).getName();
+			return new TypeGenContext(typeName, newName, this); // extend the environment with a new type environment
+		} else {
+			assert (ast instanceof wyvern.tools.typedAST.core.declarations.DefDeclaration);
+			wyvern.tools.typedAST.core.declarations.DefDeclaration methodDecl = (wyvern.tools.typedAST.core.declarations.DefDeclaration) ast;
+			String methodName = methodDecl.getName();
+			ILMethod method = new ILMethod(newName, methodDecl);
+			return new MethodGenContext(methodName, method, this); // extend the environment with a new method environment
+		}
+		
+	}
+	public abstract List<wyvern.target.corewyvernIL.decl.Declaration> genDeclSeq(); // generate the declarations, used to create a new object when declaration sequence come to the end
```

Comment:
The comments on these methods should be javadoc-style, i.e. before the declaration and using /** ... */